docs/source/contributing.md

```diff
@@ -0,0 +1,221 @@
+Please ask questions, report bugs, or request new features via GitHub issues.
+If you're new to GitHub, checking out these tips for [creating issues on GitHub](https://medium.com/nyc-planning-digital/writing-a-proper-github-issue-97427d62a20f).
+
+### Contributions to code and documentation
+We use the fork-branch-pull request (PR) workflow to add new code into Echopype.
+If you are new to this workflow, check out this [tutorial](https://medium.com/swlh/forks-and-pull-requests-how-to-contribute-to-github-repos-8843fac34ce8).
+
+We have recently moved away from Gitflow development to [trunk-based development](https://www.atlassian.com/continuous-delivery/continuous-integration/trunk-based-development) to streamline the process and reduce repo management overhead.
+The main thing to keep in mind is to set the PR target to the `main` branch in the `upstream` repository (the one sitting under the OSOceanAcoustics GitHub organization).
+We will no longer use a `dev` branch.
+
+We encourage all code contributions to be accompanied by tests and documentations when appropriate.
+We may ask for these when reviewing the PRs.
+If you have added new tests but the [GitHub Actions for continuous integration](#github-actions-for-continuous-integration-ci) need approval to run, ping the maintainers (@leewujung, @ctuguinay) to get them started.
+
+
+
+## Development environment
+
+To create an environment for developing Echopype, we recommend the following steps:
+
+1. Fork the Echopype repository following the guide above, clone your fork, then in `git remote` set your fork as the `origin` and the OSOceanAcoustics repository as `upstream`:
+    ```shell
+    # Clone your fork
+    git clone https://github.com/YOUR_GITHUB_USERNAME/echopype.git
+
+    # Go into the cloned repo folder
+    cd echopype
+
+    # Add the OSOceanAcoustics repository as upstream
+    git remote add upstream https://github.com/OSOceanAcoustics/echopype.git
+    ```
+
+2. Create a conda environment using `mamba`, and follow the steps below:
+    ```shell
+    # Create a conda environment using the supplied requirements files
+    # Note the last one docs/requirements.txt is only required for building docs
+    conda create -c conda-forge -n echopype --yes python=3.9 --file requirements.txt --file requirements-dev.txt --file docs/requirements.txt
+
+    # Switch to the newly built environment
+    conda activate echopype
+
+    # ipykernel is recommended, in order to use with JupyterLab and IPython for development
+    # We recommend you install JupyterLab separately
+    conda install -c conda-forge ipykernel
+
+    # install echopype in editable mode (setuptools "develop mode")
+    # plot is an extra set of requirements that can be used for plotting.
+    # the command will install all the dependencies along with plotting dependencies.
+    pip install -e ".[plot]"
+    ```
+
+:::{note}
+It's common to encounter the situation that installing packages using Conda is slow or fails,
```

Comment:
Maybe it's worth mentioning here that one could replace `conda` with `mamba` in the above commands to do the same thing with `mamba`

Reply:
Good point. Let me add that in.